text/0000-try-then-else-finally.md

```diff
@@ -0,0 +1,80 @@
+- Feature Name: Try, then, else, finally
+- Start Date: 2016-06-06
+- RFC PR:
+- Pony Issue: #291
+
+# Summary
+
+This RFC advocates for a rename of "then" to "finally", and introduces
+a "then" clause which can be placed immediately after the "try" block
+to program the case where no error was raised. Local variables
+introduced in the "try" block can be used here.
+
+
+# Motivation
+
+It's important to "try" only code that can actually fail, for the
+reason that it avoids making a false assumption, but also to
+communicate clearly to the reader what lines of code we're "trying".
+
+In other Pony syntax "then" means if-this-then and here it is a good
+candidate try-this-and-then. In addition, "then" is commonly used in
+promises to mean "on success". This could lead to some confusion (note
+that "then" is a reserved keyword already and cannot be used a method
+name).
+
+Instead, "finally" is used in mainstream languages such as Java and
+Python and communicates its semantic quite clearly.
+
+
+# Detailed design
+
+The "then" clause is syntax sugar that could expand to:
```

Comment:
As @Praetonus [noted in the other PR](https://github.com/ponylang/rfcs/pull/8#issuecomment-224120374), the most efficient and simplest implementation is to simply have the compiler generate the LLVM IR as if the `then` block was tacked on to the end of the `try` block, and to make sure that the Pony compiler enforces that the `then` block cannot raise an error.